crates/sncast/src/main.rs

```diff
@@ -309,8 +310,35 @@ async fn run_async_command(cli: Cli, config: CastConfig, ui: &UI) -> Result<()>
 
             let block_explorer_link =
                 block_explorer_link_if_allowed(&result, provider.chain_id().await?, &rpc, &config);
+
+            let deploy_command_message = if let Ok(response) = &result {
+                let contract_artifacts = artifacts.get(&declare.contract.clone()).ok_or(
+                    StarknetCommandError::ContractArtifactsNotFound(ErrorData {
+                        data: ByteArray::from(declare.contract.as_str()),
+                    }),
+                )?;
```

Comment:
It's a bit weird to me to have any error handling here at all. If the declare succeed then mustn't the contract exist? 
I think a simple unwrap/expect should do for all of this.

Reply:
Hmm, actually `unreachable!` would be better, considering that the contract was compiled. Will change it.